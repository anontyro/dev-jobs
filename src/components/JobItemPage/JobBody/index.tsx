import React from "react";
import styled from "@emotion/styled";
import formatDistanceToNow from "date-fns/formatDistanceToNow";
import { JobCardFooter, JobCardHeader } from "../../homePage/JobList";
import JobQueryData from "../../../interfaces/jobQueryData";

const JobItemCardHeader = styled(JobCardHeader)`
  display: flex;
  .job-card-header-body {
    display: flex;
    flex-direction: column;
    flex-grow: 1;
    h3 {
      font-size: 1.5rem;
      margin: 0;
      font-weight: bold;
    }
  }
  @media (max-width: 768px) {
  }
  @media (max-width: 425px) {
  }
`;

const JobItemCardFooter = styled(JobCardFooter)``;

const JobContentContainer = styled.div`
  .job-content-header {
  }
  .job-content-details {
    li + li:before {
      content: none;
    }
  }
`;

const dateToNow = (date: string) => {
  const toNow = formatDistanceToNow(+date);
  return `${toNow} ago`;
};

interface JobPageProps {
  job: JobQueryData;
}

const JobBody: React.FC<JobPageProps> = ({ job }) => (
  <JobContentContainer className="card job">
    <JobItemCardHeader className="card-header">
      <div className="job-card-header-body">
        <ul className="card-text-muted card-header-content">
          <li>{dateToNow(job.dateTimeAdded)}</li>
          <li>{job.type}</li>
        </ul>
        <h3>{job.title}</h3>
        <JobItemCardFooter className="card-footer">
          <span>{job.locations[0]}</span>
        </JobItemCardFooter>
      </div>
    </JobItemCardHeader>

    <div className="job-content-details">
      <p>{job.company.description}</p>
      <h3 className="bold">Requirements</h3>
      <p>{job.details.requirements.description}</p>
      <ul>
        {job.details.requirements.info.map((item: string) => (
          <li key={item}>{item}</li>
        ))}
      </ul>
    </div>
    <div className="job-content-details">
      <h3 className="bold">What You Will Do</h3>
      <p>{job.details.responsibilities.description}</p>
      <ul>
        {job.details.responsibilities.info.map((item: string) => (
          <li key={item}>{item}</li>
        ))}
      </ul>
    </div>
  </JobContentContainer>
);

export default JobBody;
