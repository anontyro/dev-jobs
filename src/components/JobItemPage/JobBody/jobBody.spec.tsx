import React from "react";
import renderer from "react-test-renderer";
import JobBody from ".";
import JobQueryData from "../../../interfaces/jobQueryData";

describe("JobBody", () => {
  it("renders correctly", () => {
    const job = {
      dateTimeAdded: "1630116426229",
      type: "Full Time",
      title: "Software Engineer",
      locations: ["London"],
      company: {
        description: "We are a small team of software engineers",
      },
      details: {
        requirements: {
          description: "We are looking for a software engineer",
          info: ["Something here"],
        },
        responsibilities: {
          description: "You will be doing this",
          info: ["Creating awesome new apps", "Using awesome new languages"],
        },
      },
    } as unknown as JobQueryData;
    const tree = renderer.create(<JobBody job={job} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
