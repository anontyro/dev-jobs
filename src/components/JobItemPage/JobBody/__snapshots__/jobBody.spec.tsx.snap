// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobBody renders correctly 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1rs61dd .job-content-details li+li:before{content:none;}",
      }
    }
    data-emotion="css 1rs61dd"
  />,
  <div
    className="card job css-1rs61dd"
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-rgejlf{font-size:0.9rem;font-family:\\"Noto Sans JP\\",sans-serif;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}.css-rgejlf .job-card-header-body{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.css-rgejlf .job-card-header-body h3{font-size:1.5rem;margin:0;font-weight:bold;}",
        }
      }
      data-emotion="css rgejlf"
    />
    <div
      className="card-header css-rgejlf"
    >
      <div
        className="job-card-header-body"
      >
        <ul
          className="card-text-muted card-header-content"
        >
          <li>
            15 days ago
          </li>
          <li>
            Full Time
          </li>
        </ul>
        <h3>
          Software Engineer
        </h3>
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": ".css-1inan0i{font-size:0.8rem;font-family:\\"Noto Sans JP\\",sans-serif;}",
            }
          }
          data-emotion="css 1inan0i"
        />
        <div
          className="card-footer css-1inan0i"
        >
          <span>
            London
          </span>
        </div>
      </div>
    </div>
    <div
      className="job-content-details"
    >
      <p>
        We are a small team of software engineers
      </p>
      <h3
        className="bold"
      >
        Requirements
      </h3>
      <p>
        We are looking for a software engineer
      </p>
      <ul>
        <li>
          Something here
        </li>
      </ul>
    </div>
    <div
      className="job-content-details"
    >
      <h3
        className="bold"
      >
        What You Will Do
      </h3>
      <p>
        You will be doing this
      </p>
      <ul>
        <li>
          Creating awesome new apps
        </li>
        <li>
          Using awesome new languages
        </li>
      </ul>
    </div>
  </div>,
]
`;
